### PR TITLE
Fix Lambda function alias name checks

### PR DESCRIPTION
**Pull request by dominikschubert**
## Motivation

A user reported an issue where invoking a lambda with an alias like `2024-01-01` would fail.

## Changes

- Introduce test covering both a valid case like `2024-01-01` and invalid one like `2024`
- Fix the alias regex we're using to check if a qualifier is an alias or not
- Add a check to the `create_alias` method. In the reported case, the alias was created but we only checked the regex on invoke which then failed to resolve the proper alias.

### CODE REVIEW

File: tests/aws/services/lambda_/test_lambda_api.py [hunk 1]
```diff
@@ -1636,6 +1636,55 @@ def test_notfound_and_invalid_routingconfigs(
             )
         snapshot.match("alias_does_not_exist_esc", e.value.response)
 
+    @markers.snapshot.skip_snapshot_verify(paths=["$..LoggingConfig"])
```
Comment (dominikschubert) [inline tests/aws/services/lambda_/test_lambda_api.py:1639]: Another reason to tackle this soonish

File: tests/aws/services/lambda_/test_lambda_api.py [hunk 2]
```diff
@@ -1636,6 +1636,55 @@ def test_notfound_and_invalid_routingconfigs(
...
+            FunctionName=function_name,
+            Handler="index.handler",
+            Code={
+                "ZipFile": create_lambda_archive(
+                    load_file(TEST_LAMBDA_PYTHON_ECHO), get_content=True
+                )
+            },
+            PackageType="Zip",
+            Role=lambda_su_role,
+            Runtime=Runtime.python3_9,
```
Comment (dfangl) [inline tests/aws/services/lambda_/test_lambda_api.py:1656]: Why python 3.9?

File: localstack/services/lambda_/api_utils.py [hunk 3]
```diff
@@ -83,7 +83,7 @@
 # Pattern for an alias qualifier
 # Rules: https://docs.aws.amazon.com/lambda/latest/dg/API_CreateAlias.html#SSS-CreateAlias-request-Name
 # The original regex from AWS misses ^ and $ in the second regex, which allowed for partial substring matches
-ALIAS_REGEX = re.compile(r"(?!^[0-9]+)(^[a-zA-Z0-9-_]+$)")
+ALIAS_REGEX = re.compile(r"(?!^[0-9]+$)(^[a-zA-Z0-9-_]+$)")
```
Comment (dfangl) [inline localstack/services/lambda_/api_utils.py:86]: I am pretty sure the last `$` in the regex is not necessary if used with match and not search, but it does not hurt either :)

**Review by dfangl — APPROVED** [review]
LGTM!

File: tests/aws/services/lambda_/test_lambda_api.py (hunk 2 above)
Comment (dominikschubert) [reply tests/aws/services/lambda_/test_lambda_api.py:1656]: copypasta

Comment (dominikschubert) [reply tests/aws/services/lambda_/test_lambda_api.py:1656]: Let's update them all when we do the LoggingConfig where we need to regenerate all the snapshots anyway

Comment (joe4dev) [reply tests/aws/services/lambda_/test_lambda_api.py:1656]: we might be able to save a lot of time on image pulling (at least in some runners not testing the multi-runtime tests) if we can make our entire test suite use the new al2023 images (so much smaller: 3.11=>3.12 from 781.95 MB =>557.9 MB).

**Pull request merged**